pkg/config/settings/http/client.go

```diff
@@ -21,15 +21,16 @@ type runtimeSettingsHTTPClient struct {
 	c                 *http.Client
 	baseURL           string
 	targetProcessName string
+	closeConnection   util.ShouldCloseConnection
 }
 
 // NewClient returns a client setup to interact with the standard runtime settings HTTP API
-func NewClient(c *http.Client, baseURL string, targetProcessName string) settings.Client {
-	return &runtimeSettingsHTTPClient{c, baseURL, targetProcessName}
+func NewClient(c *http.Client, baseURL string, targetProcessName string, closeConnection util.ShouldCloseConnection) settings.Client {
```

Comment:
for improving extensibility WDTH about adding the new parameter in a ClientOptions struct so if never later we need to add another parameter we don't need to update all the function call. 

```golang
type ClientOptions struct {
     CloseConnection util.ShouldCloseConnection
}
```

```suggestion
func NewClient(c *http.Client, baseURL string, targetProcessName string, ClientOptions{CloseConnection: util.ShouldCloseConnection}) settings.Client {
```